OpenEdXMobile/src/main/java/org/edx/mobile/http/notifications/EdxCourseAccessErrorState.kt

```diff
@@ -0,0 +1,79 @@
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private val layout: LayoutCourseAccessErrorBinding =
+        LayoutCourseAccessErrorBinding.inflate(LayoutInflater.from(context))
+
+    init {
+        this.addView(layout.root)
+        attrs?.let {
+            val typedArray = context.obtainStyledAttributes(it, R.styleable.StateLayout)
+            val state = State.values()[typedArray.getInt(R.styleable.StateLayout_state, 0)]
+            setState(state)
+        }
+    }
+
+    fun setState(state: State) {
+        when (state) {
+            State.AUDIT_ACCESS_EXPIRED -> {
+                layout.heading.text = context.getString(R.string.course_access_expired)
+                layout.description.text = context.getString(R.string.message_no_new_session)
+                layout.layoutUpgradeFeatures.root.setVisibility(false)
+                layout.primaryButton.shimmerViewContainer.hideShimmer()
+                layout.primaryButton.btnUpgrade.text = context.getText(R.string.label_find_a_course)
+                layout.primaryButton.btnUpgrade.icon = null
+                layout.secondaryButton.setVisibility(false)
+            }
+            State.IS_UPGRADEABLE -> {
+                layout.heading.text = context.getString(R.string.course_access_expired)
+                layout.description.text = context.getString(R.string.message_to_upgrade_course)
+                layout.layoutUpgradeFeatures.root.setVisibility(true)
+                layout.primaryButton.shimmerViewContainer.postDelayed({
+                    layout.primaryButton.btnUpgrade.isEnabled = false
+                    layout.primaryButton.shimmerViewContainer.startShimmer()
+                }, 250)
+                layout.secondaryButton.setVisibility(true)
+                layout.secondaryButton.text = context.getText(R.string.label_find_a_course)
+            }
+        }
+    }
+
+    fun setPrimaryButtonText(text: String) {
+        layout.primaryButton.btnUpgrade.text = text
+        layout.primaryButton.btnUpgrade.icon =
+            ContextCompat.getDrawable(context, R.drawable.ic_lock)
+        layout.primaryButton.shimmerViewContainer.postDelayed({
+            layout.primaryButton.shimmerViewContainer.hideShimmer()
+            layout.primaryButton.btnUpgrade.isEnabled = true
+        }, 500)
+    }
+
+    fun setPrimaryButtonListener(onClickListener: OnClickListener) {
+        layout.primaryButton.btnUpgrade.setOnClickListener(onClickListener)
+    }
+
+    fun setSecondaryButtonListener(onClickListener: OnClickListener) {
+        layout.secondaryButton.setOnClickListener(onClickListener)
+    }
+
+    fun enableUpgradeButton(enable: Boolean) {
+        layout.primaryButton.btnUpgrade.setVisibility(enable)
+        layout.primaryButton.loadingIndicator.setVisibility(!enable)
+    }
+
+    enum class State {
+        NONE, AUDIT_ACCESS_EXPIRED, IS_UPGRADEABLE
```

Comment:
no need to define 'NONE' state

Reply:
we are required to declare the `NONE`, cuz, in the case of a not passed state in XML, the code will automatically pick the fist enum and behave accordingly.